### ISSUE TITLE
plot_diagnostics of statsmodels

### ISSUE THREAD

**Issue by set92** [tgsmith61591/pmdarima#49]
I was checking some tutorials about arima, and I saw [one](http://www.cognoesis.com/2018/02/13/time-series-data-exploratory-data-analysis-forecast/) that it use the [plot_diagnostics](https://www.statsmodels.org/dev/generated/statsmodels.tsa.statespace.mlemodel.MLEResults.plot_diagnostics.html) to see how bad/good the model was.

I thought it could be a nice addition to pmdarima.

**Comment by charlesdrotar** [comment]
I took a stab at this using the approach you suggested:

```python
    @if_has_delegate('arima_res_')
    def plot_diagnostics(self, variable=0, lags=10, fig=None, figsize=None):
        """Get plot_diagnostics of the ARIMA model"""
        return self.arima_res_.plot_diagnostics(variable, lags, fig, figsize)
```

And executed using one of the examples form the docs:

```python
# can comment out if not using jupyter notebook
%matplotlib inline

import numpy as np
import pmdarima as pm
from pmdarima.datasets import load_wineind

# this is a dataset from R
wineind = load_wineind().astype(np.float64)

# fit stepwise auto-ARIMA
stepwise_fit = pm.auto_arima(wineind, start_p=1, start_q=1,
                             max_p=3, max_q=3, m=12,
                             start_P=0, seasonal=True,
                             d=1, D=1, trace=True,
                             error_action='ignore',  # don't want to know if an order does not work
                             suppress_warnings=True,  # don't want convergence warnings
                             stepwise=True)  # set to stepwise

stepwise_fit.plot_diagnostics(figsize=(15,12))
```

This resulted in the quirky behavior of plotting the 4 plots twice:

<img width="357" alt="screen shot 2018-11-28 at 10 33 07 pm" src="https://user-images.githubusercontent.com/3826821/49203566-a4f7a000-f35d-11e8-8814-9984440211b3.png">

So this could be solved by taking only the first 4 plots, but I wonder if we can address this at the root (i.e. force only 1 callback instead of 2)?

Here is the workaround to get just the first 4 plots. Note the addition of getting the type forces one invocation to just return the type and the other to actually plot. This is not a huge blemish in my opinion but may be worth looking into.:

```python
    @if_has_delegate('arima_res_')
    def plot_diagnostics(self, variable=0, lags=10, fig=None, figsize=None):
        """Get plot_diagnostics of the ARIMA model"""
        return type(self.arima_res_.plot_diagnostics(variable, lags, fig, figsize))
```

<img width="816" alt="screen shot 2018-11-28 at 10 50 27 pm" src="https://user-images.githubusercontent.com/3826821/49204231-0ae52700-f360-11e8-9d73-ad44688b75f3.png">

Anywho adding the PR soon and going to ref this issue to continuity purposes.

**Comment by tgsmith61591** [comment]
Thanks so much for starting this work @charlesdrotar! Something that's bothering me about this now that I think about it is that it appears the [`plot_diagnostics`](https://github.com/statsmodels/statsmodels/blob/master/statsmodels/tsa/statespace/mlemodel.py#L2584) function may only be defined for `SARIMAX` in statsmodels... that means we'd actually have to write the function rather than delegate it down. But this is a great start!

**Comment by tgsmith61591** [comment]
This was added to `develop` in #55 and will be released in v1.1.0
